src/main/java/seedu/ui/TextUi.java

```diff
@@ -211,4 +213,56 @@ public static void helpMessage() {
         printDoubleLineMessage(message);
     }
 
+    public static void confirmAddDuplicateMessage(ArrayList<Integer> duplicatedIndex, ContactList contactList) {
+        if (duplicatedIndex.size() == 1) {
+            Contact currentContact = contactList.getContactAtIndex(duplicatedIndex.get(0));
+            String message = "One of your saved contacts has a duplicate field:\n"
+                    + "\n" + duplicatedIndex.get(0) + ". " + currentContact.getName()
+                    + formatContactFields(currentContact) + "\n\nDo you still want to add the contact?  (y/n)\n";
+            printDoubleLineMessage(message);
+        } else {
+            System.out.println(LINE);
+            System.out.println("These contacts have duplicate fields:\n");
+            for (Integer index : duplicatedIndex) {
+                duplicatedContactsMessage(contactList.getContactAtIndex(index), index);
+            }
+            System.out.println("Do you still want to add the contact?  (y/n)\n");
+            System.out.println(LINE);
+        }
+    }
+
+    public static void confirmEditDuplicateMessage(ArrayList<Integer> duplicatedIndex, ContactList contactList) {
+        if (duplicatedIndex.size() == 1) {
+            Contact currentContact = contactList.getContactAtIndex(duplicatedIndex.get(0));
+            String message = "One of your saved contacts has a duplicate field:\n"
+                    + "\n" + duplicatedIndex.get(0) + ". " + currentContact.getName()
+                    + formatContactFields(currentContact) + "\n\nDo you still want to edit the contact?  (y/n)\n";
+            printDoubleLineMessage(message);
+        } else {
+            System.out.println(LINE);
+            System.out.println("These contacts have duplicate fields:\n");
+            for (Integer index : duplicatedIndex) {
+                duplicatedContactsMessage(contactList.getContactAtIndex(index), index);
+            }
+            System.out.println("Do you still want to edit the contact?  (y/n)\n");
+            System.out.println(LINE);
+        }
+    }
```

Comment:
It would appear that these two methods are largely similar, except for one line asking for confirmation w.r.t. `add` or `edit`. My suggestion is to:
- Combine into a single method `confirmDuplicateMessage`
- Include one more parameter `String type` which allows you to decipher if the operation was `add` or `edit`
- Use an `if-else` block to decide what to print, for instance:
```java
if (type.equals("add") {
    System.out.println("Do you still want to add the contact?");
} else {
    System.out.println("Do you still want to edit the contact?");
}
```
- When calling this method in `AddContactCommand` or `EditContactCommand`, include your specified `type`, for example:
```java
TextUi.confirmDuplicateMessage(duplicatedIndex, contactList, "add"); // for add
TextUi.confirmDuplicateMessage(duplicatedIndex, contactList, "edit"); // for edit
```